xlators/cluster/afr/src/afr-common.c

```diff
@@ -3444,7 +3464,7 @@ afr_lookup_selfheal_wrap(void *opaque)
 
     ret = afr_selfheal_name(frame->this, pargfid, local->loc.name,
                             &local->cont.lookup.gfid_req, local->xattr_req);
-    if (ret == -EIO)
+    if ((ret == -EIO) && !afr_is_sbrain_resolution_lookup(frame, local))
```

Comment:
afr_gfid_split_brain_source()  is called by the time control reaches here. If it still fails, why should afr treat it as success?

Reply:
@pranithk  We need lookup to succeed for glfsheal sbrain resolution as mentioned in the commit message.

Reply:
> mentioned in the commit message.

Only in case you were wondering, updated/force-pushed PRs with different commit messages don't update what you see in the first comment in the page. We need to click on the Commits tab and then click on the patch to see the actual commit message.

Reply:
I saw the commit message, but I am not understanding why lookup should succeed despite glfsheal not able to figure out source based on the policy set in heal-op.

Reply:
Ah, that is because glfsh_heal_splitbrain_file() first does lookup to fill the loc. Only if it succeeds, it goes to the next step of healing the xattrs via the overloaded gexattr.

Reply:
Okay, so in retry1 glfs_resolve() should succeed so that syncop_lookup() can fix the gfid. Did I get that right? In this case
glfs_resolve() should succeed despite being in gfid split-brain but syncop_lookup() can fail if the gfid-split-brain resolution doesn't happen as intended. Is the xattr_req checking of the "heal-op" necessary?

Reply:
No, retry1+ glfs_resolve()  is for the parent path, which will anyway succeed since its not in split-brain. 
syncop_lookup() is for the actual file that is in split-brain, which must succeed, so that the next step of syncop_getxattr() can try to heal (and fail if the chosen policy is not satisfied) the split-brain.

For the test `EXPECT "No bigger file for file /test/f3" $CLI volume heal $V0 split-brain bigger-file /test/f3`, glfs_resolve is for `/test` and syncop_lookup() is for `f3`

Reply:
I think it should be handled in cli code. lookup should fail if it can't decide which one is the source. It is wrong to serve success if it doesn't know which one is source at the time of serving lookup. CLI seems to print lookup-failed and then no-bigger-file etc. Maybe you should change that part to not print lookup-failed if gfid-heal-msg is present.

Reply:
Done

Reply:
@pranithk Please review/merge if patch is ok.